instrumentation/kafka-streams/src/main/java/brave/kafka/streams/KafkaStreamsTracing.java

```diff
@@ -42,18 +42,26 @@
 /** Use this class to decorate Kafka Stream Topologies and enable Tracing. */
 public final class KafkaStreamsTracing {
 
-  final Tracing tracing;
+  final KafkaTracing kafkaTracing;
   final TraceContext.Extractor<Headers> extractor;
   final TraceContext.Injector<Headers> injector;
 
   KafkaStreamsTracing(Builder builder) { // intentionally hidden constructor
-    this.tracing = builder.tracing;
-    this.extractor = tracing.propagation().extractor(KafkaStreamsPropagation.GETTER);
-    this.injector = tracing.propagation().injector(KafkaStreamsPropagation.SETTER);
+    this.kafkaTracing = builder.kafkaTracing;
+    this.extractor = kafkaTracing.tracing().propagation().extractor(KafkaStreamsPropagation.GETTER);
+    this.injector = kafkaTracing.tracing().propagation().injector(KafkaStreamsPropagation.SETTER);
   }
 
   public static KafkaStreamsTracing create(Tracing tracing) {
-    return new KafkaStreamsTracing.Builder(tracing).build();
+    return new KafkaStreamsTracing.Builder(KafkaTracing.create(tracing)).build();
+  }
+
+  public static KafkaStreamsTracing create(KafkaTracing kafkaTracing) {
+    return new KafkaStreamsTracing.Builder(kafkaTracing).build();
+  }
+
+  public KafkaTracing kafkaTracing() {
```

Comment:
this doesn't need to be exposed, but I'll scrub it out later. We should document when things are exposed. The only reason we expose in KafkaTracing is for this component. If we expose something public, there should be a use case for it.

Reply:
edf2b1b1952233bfad5363fd2d93be8b835a567d